fineract-provider/src/main/java/org/apache/fineract/infrastructure/entityaccess/service/FineractEntityAccessReadServiceImpl.java

```diff
@@ -220,30 +197,35 @@ public FineractEntityAccessData mapRow(final ResultSet rs, @SuppressWarnings("un
     public String getSQLQueryInClauseIDList_ForLoanProductsForOffice(Long officeId, boolean includeAllOffices) {
 
         FineractEntityType firstEntityType = FineractEntityType.OFFICE;
-        FineractEntityAccessType accessType = FineractEntityAccessType.OFFICE_ACCESS_TO_LOAN_PRODUCTS;
-        FineractEntityType secondEntityType = FineractEntityType.LOAN_PRODUCT;
+        
+        FineractEntityRelation fineractEntityRelation = fineractEntityRelationRepository
+				.findOneByCodeName(FineractEntityAccessType.OFFICE_ACCESS_TO_LOAN_PRODUCTS.toStr());
+		Long relId = fineractEntityRelation.getId();
 
-        return getSQLQueryInClause_WithListOfIDsForEntityAccess(officeId, firstEntityType, accessType, secondEntityType, includeAllOffices);
+        return getSQLQueryInClause_WithListOfIDsForEntityAccess(firstEntityType, relId, officeId, includeAllOffices);
     }
 
     @Override
     public String getSQLQueryInClauseIDList_ForSavingsProductsForOffice(Long officeId, boolean includeAllOffices) {
 
         FineractEntityType firstEntityType = FineractEntityType.OFFICE;
-        FineractEntityAccessType accessType = FineractEntityAccessType.OFFICE_ACCESS_TO_SAVINGS_PRODUCTS;
-        FineractEntityType secondEntityType = FineractEntityType.SAVINGS_PRODUCT;
+        FineractEntityRelation fineractEntityRelation = fineractEntityRelationRepository
+				.findOneByCodeName(FineractEntityAccessType.OFFICE_ACCESS_TO_SAVINGS_PRODUCTS.toStr());
+		Long relId = fineractEntityRelation.getId();
+        
 
-        return getSQLQueryInClause_WithListOfIDsForEntityAccess(officeId, firstEntityType, accessType, secondEntityType, includeAllOffices);
+        return getSQLQueryInClause_WithListOfIDsForEntityAccess(firstEntityType, relId, officeId, includeAllOffices);
     }
 
     @Override
     public String getSQLQueryInClauseIDList_ForChargesForOffice(Long officeId, boolean includeAllOffices) {
 
         FineractEntityType firstEntityType = FineractEntityType.OFFICE;
-        FineractEntityAccessType accessType = FineractEntityAccessType.OFFICE_ACCESS_TO_CHARGES;
-        FineractEntityType secondEntityType = FineractEntityType.CHARGE;
-
-        return getSQLQueryInClause_WithListOfIDsForEntityAccess(officeId, firstEntityType, accessType, secondEntityType, includeAllOffices);
+        FineractEntityRelation fineractEntityRelation = fineractEntityRelationRepository
+				.findOneByCodeName(FineractEntityAccessType.OFFICE_ACCESS_TO_CHARGES.toStr());
+		Long relId = fineractEntityRelation.getId();
```

Comment:
There is a chance of null pointer exception if FineractEntityAccessType.OFFICE_ACCESS_TO_CHARGES doesn't exists